src/services/sync/strategies/WebUntis/WebUntisBaseSyncer.js

```diff
@@ -0,0 +1,781 @@
+const { BadRequest } = require('@feathersjs/errors');
+const Syncer = require('../Syncer');
+
+const assert = require('assert');
+const axios = require('axios');
+const cookie = require('cookie');
+
+/**
+ * Type of entities.
+ * @enum {number}
+ */
+const EntityType = {
+	CLASS: 1,
+	TEACHER: 2,
+	SUBJECT: 3,
+	ROOM: 4,
+	STUDENT: 5,
+};
+
+const DayMapping = {
+	1: 'Sonntag',
+	2: 'Montag',
+	3: 'Dienstag',
+	4: 'Mittwoch',
+	5: 'Donnerstag',
+	6: 'Freitag',
+	7: 'Samstag',
+};
+
+
+/**
+ * Implements helper functions for WebUntis Syncer by means of a superclass.
+ * @class WebUntisBaseSyncer
+ * @implements {Syncer}
+ */
+class WebUntisBaseSyncer extends Syncer {
+	/**
+	* Constructor
+	*
+	* @param {*} app
+	* @param {*} stats
+	* @param {*} logger
+	* @param {*} data
+	*/
+	constructor(app, stats, logger, data) {
+		super(app, stats, logger);
+		this.data = data;
+
+		this.account = account;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['account' is not defined.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)